caps/management/commands/process_scottish_reporting_data.py

```diff
@@ -0,0 +1,428 @@
+                    point_type=point_type,
+                    point_data=row[scope],
+                    scope=scope,
+                    units=units,
+                )
+            if not pd.isna(row["Total"]):
+                point_type = "overall emissions ({})".format(row["Year"])
+                self.make_data_point(
+                    point_type=point_type, point_data=row["Total"], units=units
+                )
+
+        return 1
+
+    def extract_emissions_sources_from_sheet(self, sources_df):
+        for index, row in sources_df.iterrows():
+            point_type = row["Emission source"]
+            point_data = row["Emissions (tCO2e)"]
+            if point_type in [
+                "Please select from drop down box",
+                "Other (please specify in comments)",
+            ]:
+                continue
+            self.make_data_point(
+                point_type="emissions source",
+                point_data=point_data,
+                sub_type=point_type,
+                scope=row["Scope"],
+                units="tCO2e",
+            )
+
+        return 1
+
+    def extract_targets_from_sheet(self, targets_df):
+        type_header = "Name of Target"
+        if type_header not in targets_df.columns:
+            type_header = "Name of target"
+
+        for index, row in targets_df.iterrows():
+            point_type = row[type_header]
+            point_data = row["Target"]
+            point_units = row["Units"]
+            year = row["Target completion year"]
+            if point_data in [
+                "Please select from drop down box",
+                "Other (please specify in comments)",
+            ]:
+                continue
+            self.make_data_point(
+                point_type="target",
+                point_data=point_data,
+                sub_type=point_type,
+                units=point_units,
+                target_year=year,
+            )
+
+        return 1
+
+    def extract_projects_from_sheet(self, targets_df):
+        """
+        we don't replace the columns with our own as sadly they aren't always in the
+        same order so instead we have to do some comparison to work out which column to use.
+        Note that some of them are the same, only with spaces at the end.
+        """
+        measurement_columns = [
+            "Are these savings figures estimated or actual? ",
+            "Are these savings figures estimated or actual?",
+            "Savings figures are estimated or actual ",
+            "Savings figures are estimated or actual",
+        ]
+
+        measurement = "Savings figures are estimated or actual"
+        for col in measurement_columns:
+            if col in targets_df.columns:
+                measurement = col
+
+        for col in [
+            "Primary fuel/emission source saved",
+            measurement,
+            "First full year of CO2e savings",
+        ]:
+            targets_df[col] = targets_df[col].replace(
```

Comment:
Don't do this, but fyi you could set this as an NA value on import and then use normal NA fill or processing.

```
read_csv(..., na_values = ["Please select from drop down box"])
```